lib/multi-beacon-node.ts

```diff
@@ -0,0 +1,37 @@
+import {Chain, ChainOptions, defaultChainOptions, DrandNode, HealthCheckResponse} from './index'
+import HttpCachingChain from './http-caching-chain'
+
+class MultiBeaconNode implements DrandNode {
+    constructor(public baseUrl: string, private options: ChainOptions = defaultChainOptions) {
+    }
+
+    async chains(): Promise<Array<Chain>> {
+        const response = await fetch(`${this.baseUrl}/chains`)
+        const chains = await response.json()
+
+        if (!Array.isArray(chains)) {
+            throw Error(`Expected an array from the chains endpoint but got: ${chains}`)
+        }
```

Comment:
I think this doesn't support exposing only the `default` chain on the `dns.tld/public/latest` URLs, without chain hashes. Which I think a relay could choose to do. 
See:
- https://github.com/drand/drand/blob/a2a22571f4d0b4c76587fa4bd4bda4c95d31a848/cmd/relay/main.go#L86-L99
- https://github.com/drand/drand/blob/a2a22571f4d0b4c76587fa4bd4bda4c95d31a848/cmd/relay/main.go#L139-L142

Reply:
aha interesting - I knew this was one possible endpoint but I didn't realise a relay can host _just_ default like that.
Can the opposite also be true? Can a relay _only_ expose `/{chainId}/public` and not `/public`?

Reply:
it doesn't look like it from the code and I don't think our spec is tight enough to define it rn

Reply:
Yes, the relay can decide to host a single chainHash, or will default to the "default" one, if the flag is not set:

https://github.com/drand/drand/blob/a2a22571f4d0b4c76587fa4bd4bda4c95d31a848/cmd/relay/main.go#L73-L81

Reply:
`/public` is exposed only if the `default` chainHash is exposed.

Reply:
Nope!

Reply:
I'd suggest we have a separate interface for relays then - perhaps their spec will evolve differently to nodes

Reply:
cloudflare and PL seem to host stuff at the default path and I imagine they'll make up most of the relay usage for now - perhaps push this to a later thought?

Reply:
done